### ISSUE TITLE
No LifeTag detections from dual mode SGs

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#836]
Recently I've had a couple test deployments running of dual-mode SG's ([7399 ](https://motus.org/data/receiverDeployment?id=7399)and [5926](https://motus.org/data/receiverDeployment?id=5926)) and found that LifeTag detections are not coming through into the database, despite showing up in the Web Interface. 

I don't believe this was an issue previously so perhaps there has been some change that is creating a backlog?

**Comment by Hussell** [comment]
Which tags are you expecting to see?

**Comment by jsayers-bsc** [comment]
> 
> Which tags are you expecting to see?

Well I thought that I had been seeing several of the test tags we've had here with on-going deployments, and which I can see in the Web Interface. But looking at the receivers they've been detected by, there are **no** SG's in there; just SS.

Is this related to data transfer backlogs, such as had been reported back in #658?

**Comment by denislepage** [comment]
Yup. If you are talking about ctt tags on SG dual mode stations, I haven’t heard a confirmation from CTT yet that the processing has started.

**Comment by StuMackenzie** [comment]
Does it need to be processed by them first?

**Comment by denislepage** [comment]
They have all the individual hits, and they need to build a script to import them and filter for the Motus tags. I will ask next time we chat, or I can poke Bob and Jessica.

**Comment by StuMackenzie** [comment]
In process. 

We need to update software on all SG- dual mode receivers to the newest software. Need to find the list of dual-mode SG. @jsayers-bsc 

Catherine to compare Metadata from dual-mode receivers to the data. @cjardine-bsc 

Once data from 'old' dual-mode SG's, has been integrated by @denislepage , we should also look at these data to ensure it looks OK.

A little more in https://motuswts.slack.com/archives/C01JJ4PKQCU/p1611687686007600

**Comment by adamdsmith** [comment]
Is this fixed software available? I've been using this for Pi-based SGs:

https://s3.amazonaws.com/media.celltracktech.com/sensorgnome/raspberry/rpi_sensorgnome_2019-10-01_ctt_update.tar.bz2

**Comment by StuMackenzie** [comment]
@jsayers-bsc and @leberrigan and @dfloer are better to respond. 

Where are clear instructions for users available?

**Comment by dfloer** [comment]
I can add that patch to an existing install image, but I'd like to make sure it's tested before we release it for a wider audience.

Is there any reason to have separate images for dual-mode receivers? Is there any downside to running the same image on a receiver that doesn't also have CTT dongles?

**Comment by denislepage** [comment]
It was just a patch that CTT compiled and made available. This will likely need a bit of coordination, but the fewer version we have to support the better.

The functionality that version introduced  to stream CTT detections (and GPS) separately should be incorporated into our current branch.

They also made other changes to work on the Sensors ration, so we should plan to coordinate future updates with them.

**Comment by denislepage** [comment]
Made some progress on the import part. Just need a couple follow-up from Bob.

Couple loops to make sure are closed.

Manually uploading files from the newer software ((gz files separate from pulse data for lotek tags), should be streamed to an upload folder and then push to AWS.

Once the process is up and running, we should make sure the data come back and go into det.

We should also narrow down the list of receivers we have to manually scan that are running the old software based on when we last saw CTT tags on them. Can any of those be updated? Can we then deprecate the old software. This will have to wait until we have the import set up though, or look through the 100,000+ files we pushed to AWS

Maybe worth following up on the other dual modes we have in metadata and see if there are CTT data coming back for those once the import is up and running.

**Comment by dfloer** [comment]
Existing SG image with the CTT patch applied to it: https://beta.sensorgnome.org/sg-releases/BETA-SGPI_2021-01-26_LIWIXI_CTT.zip
I've tested to make sure it boots, and it appears to work, but I'd like it to be more tested before we tell other people to use it.

(That link also might stop working at some point, I'm still building the website.)

**Comment by StuMackenzie** [comment]
Further documentations and notes in https://github.com/MotusDev/Motus-TO-DO/issues/658

**Comment by adamdsmith** [comment]
I'll try to test this image in the coming weeks with both Lotek and CTT tags. 

Got a quick summary of what you're looking for from a performance stand point? I suppose we expect both tags to show up as live pulses via the web interface, or maybe that was already happening? If it's about tags lost during data processing, I can submit the raw data files here for your perusal.

**Comment by denislepage** [comment]
We have been on hold with CTT for a while for them to process dual mode files. We are almost there. The functionality on the receiver should be working. The key with the new image is making sure that CTT tags are streamed to separate files, and that both CTT and lotek pulse files include gps readings.

**Comment by jsayers-bsc** [comment]
@adamdsmith Yes If the CTT patch is successfully applied you will see CTT tags in the Live Known Tags pane of the Web Interface -- but only if there is already a FCD or other compatible dongle plugged in (since that entire pane only shows up when a 166 dongle is plugged in).  And you're also likely to see a few false positives since it doesn't rely on a tag database being loaded on the SG as with 166 tags.

And to clarify, the CTT compatibility patch can be applied as part of the initial install just by properly renaming it and dropping it in the `uboot` folder of the zip. My local copy of the RPi SG software has it in there so any time I make  new SD card it is already CTT compatible. @dfloer and @adamdsmith I think there was some confusion over the word "image" since the RPi SG is just the files inside the .zip

As Denis mentioned there are still some backlogs on the processing end but making an RPi SG dual-mode is very simple and just requires the patch (the link @adamdsmith posted is the one I've been using).

**Comment by dfloer** [comment]
@adamdsmith This is just the CTT patch and other updated applied to the last base SG image. There isn't anything new here. I just put it together so that we can just have the one thing to worry about and give people if needed.

It should work, but I may have messed up on permissions or something and want to make sure I didn't and that it works as expected.

**Comment by adamdsmith** [comment]
I'm not seeing the CTT motus tag adapter show up anywhere on the web interface, either under `Devices` or `what I'm doing now...`. Is it supposed to show up anywhere? This is using both of the links I and @dfloer provided above. Also, CTT tags will only show up in the "Live Known Tags" pane, and not the "Live pulses" pane? So I need to install the tag database that contains registered CTT tags?

**Comment by jsayers-bsc** [comment]
I'll give this combined version a shot tomorrow.

I do know that for those where the patch is applied separately that 
- often 2 reboot cycles are needed
- the CTT dongle will show up in Devices, but not in What I'm doing now (image below)
-

**Comment by jsayers-bsc** [comment]
First glance suggests the combined version with the patch works. I'm seeing everything displayed as it should be, including live CTT tags (once a FCD has been attached)

**Comment by adamdsmith** [comment]
Cool. Would you mind creating a fresh link to a zipped directory that
replicates what's on the microSD card of your functional unit(s)? THanks!

**Comment by jsayers-bsc** [comment]
I just copied over everything from the .zip Dale provided. Took several attempts to download (not sure why) but after that it just went normally.

**Comment by adamdsmith** [comment]
OK, I tried that version too but will try again. Are you using the newest CTT Motus dongle, or the older version with built in USB cable ([like the one pictured on their [](website](https://celltracktech.com/products/tag-system/lifetag-motus-adapter/))?) The newer version requires the user to purchase the USB-A to microB adapter to connect to the SG. I don't know if there are particular specs needed for the cable (I'm using a USB 2.0 version). In any case, I'll check back here after trying again...

**Comment by jsayers-bsc** [comment]
Hmm... I guess these would be the first version then. 

I haven't seen the newer version. The need for an adapter is a bit of a pain. Wonder how many of the V1 adapters they have left? :)

**Comment by adamdsmith** [comment]
Just retried with @dfloer "image." Still no luck getting it to show up. It's not the port on the Pi; FUNcube recognized without problem. Tried multiple cables. Something is amiss. Here's what the newest version looks like...

<img src="https://user-images.githubusercontent.com/3195855/107660390-33fae500-6c56-11eb-9eb6-a02dd641ac20.jpg" width="400" />

<img src="https://user-images.githubusercontent.com/3195855/107660404-38bf9900-6c56-11eb-9ce5-68e44a9405aa.jpg" width = "400" />

And yes, it's annoying to have to purchased the USB cables separately. Especially when you aren't expecting that to be the case...

**Comment by adamdsmith** [comment]
FWIW, now that I think about it, I tried an earlier version of this patch on a couple of SGs with the V1 Motus dongle and it recognized the dongles just fine. Indeed, one of them is here: https://sg-433crpi3b3b3.sensorgnome.org/

**Comment by jsayers-bsc** [comment]
There is a dual-mode SG now running at HQ with 2 LifeTags (with a V1 CTT-Motus dongle). It's networked so should be syncing data. Sounds like processing dual-mode data is still not entirely implemented on the CTT end, but I think I'll just leave this one on and syncing so we can confirm when everything is properly flowing.

**Comment by dfloer** [comment]
@adamdsmith Thanks for looking into that. Guess I need to get one of the new dongles to poke at, it's probably something similar like USB IDs have changes.

**Comment by StuMackenzie** [comment]
@adamdsmith in the meantime we can send you some of the older dongles if you'd like to try those instead while we're mucking about this is?

**Comment by adamdsmith** [comment]
Thanks! I think I'm ok for the short term... I'll simply hope the fix is straightforward.

**Comment by adamdsmith** [comment]
Hey @dfloer @StuMackenzie 

Not being pushy, honest (!), just curious if there's any expected timeline for exploring this issue? If not, I may think harder about taking you up on the offer for a few V1 CTT dongles. Thanks!

**Comment by jsayers-bsc** [comment]
Hi Adam, at the moment there are no CTT detections available for dual-mode SG's, even those where the dongle is recognized and recording properly. I believe the issue is that the data flow from the motus server, to CTT, and back again hasn't been fully implemented yet. I _think_ the hold up at the moment is on the CTT end, but am not sure. It may just require another poke. @denislepage would know best. We do have multiple dual-mode SG's deployed already so it would be great to close the loop on data processing.

**Comment by adamdsmith** [comment]
Thanks @jsayers-bsc. Good to know re: processing of dual-mode SGs; I wasn't aware it was not fully implemented. 

More specifically, though, I was referring to the V2 dongles not being acknowledged on the web interface. If we're confident they're still recognized by the SG software and recording, that's fine. I'm simply not sure if that's the case.

**Comment by jsayers-bsc** [comment]
> More specifically, though, I was referring to the V2 dongles not being acknowledged on the web interface. If we're confident they're still recognized by the SG software and recording, that's fine. I'm simply not sure if that's the case.

Yeah that's a good point. I wouldn't feel very comfortable with it if I couldn't confirm they are recognized in the web interface. I guess you could check the raw data files to see if CTT data is being recorded, but I'd be very surprised if they are recording without the dongles being reported. You do have 166 MHz dongles plugged in too right? I do know that the CTT dongles won't show up if the "Live Pulses" and "Devices" panes aren't already displayed from having a FCD or similar plugged in.

In the meantime if you do need some V1s to get out so that they are recording we can send some.

**Comment by adamdsmith** [comment]
Thanks. Yes, FCD are attached and detected/reported properly but not so with V2 dongles. As you reported earlier, I do "see" V1 dongles on the interface. If you have a few V1 dongles you can spare, that would be handy. I hope to get at a single CTT dongle out on at least three SGs this spring (upgrading each with a 434 MHz omni), COVID-dependent...

**Comment by jsayers-bsc** [comment]
Sure, 3 dongles then?

**Comment by adamdsmith** [comment]
Yeah, that'd be great. I'll send you an email...

**Comment by denislepage** [comment]
This is indeed sitting at CTT’s end.

CTT has all the data and we have the scripts ready to go as soon as they start pushing us the data.

**Comment by adamdsmith** [comment]
@jsayers-bsc Got the V1 CTT dongles and can confirm they work as you describe above. So obviously some (hopefully) minor change on the V2 is preventing them being recognized. Thanks for sending those. Arrived just in time as they'll go out this weekend.

I'm also having trouble accessing @dfloer patch above, something about invalid credentials.

**Comment by jsayers-bsc** [comment]
Excellent.

Also I confirmed with CTT that the V1 dongles are out of stock and no longer being produced. We have a few V2 dongles en route to us here (one will go to Dale). Hopefully your issue with them is just a weird fluke because that's all we have to work with now for dual mode SGs!

**Comment by dfloer** [comment]
@adamdsmith Whoops, forgot that beta. was using the same cert as sensorgnome.org and that I'd need to update that. I'll fix that. In the mean time, you can tell your browser that it's fine and still get the file.

**Comment by adamdsmith** [comment]
Thanks @dfloer. For whatever reason it won’t give me the option to except it from the certificate issue, across multiple browsers.

**Comment by dfloer** [comment]
I've fixed the certificate for now. Fair warning, this is also a _beta_ website too, so it should work, but may not while I'm messing with things. Like I'm using this as an opportunity to update the server's software.

**Comment by adamdsmith** [comment]
Hey @dfloer @jsayers-bsc 

Any progress on this SG software front for dual-mode units? I'm not having any luck with SensorGnomes detecting the V2 CTT adapters.

**Comment by jsayers-bsc** [comment]
@dfloer can clarify but I believe he found the V2 CTT adapter to be incompatible with SG currently as the output format is different. I'm not sure what options we have on the SG end for supporting it in its current form; the ideal solution seems to be for CTT to stick with the same data format as V1, which is recognized by the SG. Dale, please correct any of that if I got the details wrong. I have opened a support ticket with CTT, but haven't yet heard back beyond the initial confirmation.

**Comment by dfloer** [comment]
@adamdsmith @jsayers-bsc I got the V2 receiver late last week, and haven't had a chance to take more than a cursory look at it.

It's using a different microcontroller, so different USB IDs. But the data format it's emitting has changed. The V1s just emitted the tag ID and the RSSI: `61343407,-84` on the serial port. The new ones spit out JSON: `{"protocol":"1.0.0","meta":{"data_type":"coded_id","rssi":-84},"data":{"id":"61343407"}}`.
The practical ramifications of this is that the Sensorgnome code will need to be changed to be able to read the new receivers, as it's not just a drop-in replacement.

I'll look more into this later this week or next.

**Comment by adamdsmith** [comment]
Thanks @dfloer.  That’s pretty annoying. I really appreciate you looking into it.

**Comment by StuMackenzie** [comment]
If you need some V1 dongles in the meantime we may have some we could send you.

____________________

**Comment by jsayers-bsc** [comment]
@adamdsmith You may also want to contact / open a support ticket with CTT directly. I've already done so, but the more people that bring it up with them the more likely this gets resolved, at least for future dongles. Once we're out of V1 dongles, which won't be very long, we won't be able to deploy CTT-compatible SensorGnomes.

**Comment by adamdsmith** [comment]
Good point. I’m sitting on a stack of useless V2 dongles at the moment. Thanks for the offer of additional V1, but I’m not deploying anything for a couple months. Did you put in the ticket somewhere specific? Otherwise I’ll just contact David. Thanks.

**Comment by StuMackenzie** [comment]
It’s as much on us to make sensorgnomes compatible as it is on them to fix the dongle. It may be easier for us to fix sg than them to fix the dongle.

____________________

**Comment by jsayers-bsc** [comment]
True, though to be fair the manufacturer is currently selling a product that does not work as advertised -- not because the SG has changed, but because they modified their product.

**Comment by adamdsmith** [comment]
You're both right. :) In any case, I've sent them another note regarding the change in functionality from V1 to V2...

**Comment by adamdsmith** [comment]
CTT update: they suggested the V2 dongles were "working here in-house" but will apparently get Bob to look into it. Suggested switching over the SensorStations as the best fix...

**Comment by dfloer** [comment]
Taking a look at this now. First thoughts on the code are that I'll need to change the udev rules to pick up the new USB device (the microcontroller) and need the change the parsing from the raw serial.

Slight niggle that I'm coming across trying to prototype the second one, sometimes the messages are getting cut off at 64B (which is the maximum data per packet in USB 2.0), which is gonna cause some headaches with parsing. As far as I can tell, the messages are getting sent still, so I'll need to figure out what's going on.

**Comment by adamdsmith** [comment]
Hey @dfloer and others, any progress on SGs working with the V2 CTT dongles? I'm really sorry to come back to this, but I'll be converting several stations to dual-mode in the next month or two, and I need to decide soon whether adding V2 dongles to existing SGs will be possible or if I need to convert a select few of them to SensorStations.

**Comment by jsayers-bsc** [comment]
> Hey @dfloer and others, any progress on SGs working with the V2 CTT dongles? I'm really sorry to come back to this, but I'll be converting several stations to dual-mode in the next month or two, and I need to decide soon whether adding V2 dongles to existing SGs will be possible or if I need to convert a select few of them to SensorStations.

I'm not sure where this stands but I don't think we expect a solution imminently. Recently I tested a V2 dongle on a SensorStation (in one of the USB ports) and from what I can tell no data is being recorded on that port. So the issue with the V2 dongle may go beyond SG compatibility. 

We do still have some V1 here that we could send as temporary solution. Depends what your timeline is, and how many you need.

**Comment by StuMackenzie** [comment]
Closing/Resolved? The V2 issue was resolved by CTT, but I'm not sure that we've followed up in the best way with users? 

patch and instructions on using V2 dongles is publicly available somewhere?
